ssl/statem/statem_srvr.c

```diff
@@ -2968,6 +3070,161 @@ static int tls_process_cke_psk_preamble(SSL *s, PACKET *pkt)
 #endif
 }
 
+#ifndef OPENSSL_NO_CNSM
+static int tls_process_cke_sm2ecc(SSL *s, PACKET *pkt)
+{
+    unsigned char rand_premaster_secret[SSL_MAX_MASTER_KEY_LENGTH];
+    size_t decrypt_len = 0;
+    PACKET enc_premaster;
+    int ret = 0;
+    EVP_PKEY *pkey = NULL;
+    
+    EVP_PKEY_CTX *pkey_ctx = NULL;
+
+    pkey = s->cert->pkeys[SSL_PKEY_ECC_ENC].privatekey;
+    if (pkey == NULL) {
+        SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_TLS_PROCESS_CKE_SM2ECC,
+                 SSL_R_MISSING_RSA_CERTIFICATE);
+        goto err;
+    }
+
+    /* SSLv3 and pre-standard DTLS omit the length bytes. */
+    if (s->version == SSL3_VERSION || s->version == DTLS1_BAD_VER) {
+        enc_premaster = *pkt;
+    } else {
+        if (!PACKET_get_length_prefixed_2(pkt, &enc_premaster)
+            || PACKET_remaining(pkt) != 0) {
+            SSLfatal(s, SSL_AD_DECODE_ERROR, SSL_F_TLS_PROCESS_CKE_SM2ECC,
+                     SSL_R_LENGTH_MISMATCH);
+            goto err;
+        }
+    }
+			
+		/*
+     * We must not leak whether a decryption failure occurs because of
+     * Bleichenbacher's attack on PKCS #1 v1.5 RSA padding (see RFC 2246,
+     * section 7.4.7.1). The code follows that advice of the TLS RFC and
+     * generates a random premaster secret for the case that the decrypt
+     * fails. See https://tools.ietf.org/html/rfc5246#section-7.4.7.1
+     */
+
+    if (RAND_priv_bytes(rand_premaster_secret,
+                      sizeof(rand_premaster_secret)) <= 0) {
+        SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_TLS_PROCESS_CKE_SM2ECC,
+                 ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
```

Comment:
This doesn't actually seem to be properly used, you just decrypt to that buffer.

Reply:
Yes, it's really not necessary. I just copy the part implement of tls_process_cke_rsa() , and don't check the purpose of those code carefully.